Source/modules/webcl/WebCLContext.idl

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2015 Intel Corporation All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+typedef unsigned long CLenum;
+typedef unsigned long CLuint;
+typedef boolean CLboolean;
+[
+    Conditional=WEBCL,
+    NotScriptWrappable
+] interface WebCLContext {
+    [RaisesException] WebCLBuffer createBuffer(CLenum memFlags, ImageData hostPtr);
+    [RaisesException] WebCLBuffer createBuffer(CLenum memFlags, HTMLCanvasElement hostPtr);
+    [RaisesException] WebCLBuffer createBuffer(CLenum memFlags, HTMLImageElement hostPtr);
+    [RaisesException] WebCLBuffer createBuffer(CLenum memFlags, CLuint sizeInBytes, optional ArrayBufferView hostPtr);
+    [RaisesException] WebCLCommandQueue createCommandQueue(optional WebCLDevice? device, optional CLenum properties);
+    // TODO(junmin-zhu): Blink does not support WebIDL dictionary definitions currently.
+    // So it requires custom binding to get properties of WebCLImageDescriptor. At the same time,
+    // it's impossible to separate same idl functions w/ and w/o custom binding. So implemented through custom binding.
+    // [Custom, RaisesException] WebCLImage createImage(CLenum memFlags, ImageData hostPtr);
+    // [Custom, RaisesException] WebCLImage createImage(CLenum memFlags, HTMLCanvasElement hostPtr);
+    // [Custom, RaisesException] WebCLImage createImage(CLenum memFlags, HTMLImageElement hostPtr);
+    [Custom, RaisesException] WebCLImage createImage(CLenum memFlags, WebCLImageDescriptor descriptor,
+                                                     optional ArrayBufferView hostPtr);
```

Comment:
It includes "PermissiveDictionaryConversion" extended attributes. It requires to back port upstream code, which modify the "IDLExtendedAttributes.txt", "scripts/v8_dictionary.py" and "scripts/v8_methods.py". 

I suggest to do this in next rebasing.